c10/fp16/bitcasts.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
```

Comment:
Can we (1) namespace the functions in this file and (2) prefix the macros with `C10_`? It's good header hygiene. Also, since you have `#pragma once` you don't also need an ifdef guard.

Reply:
Also, lets put this in `core`, not in an entirely new folder (that will only have one file, for the forseeable future.) This will also save you from having to make some build system changes (you have to do extra work when you add a new folder.)